### PR TITLE
Fix parsing of lowercase S/W directions

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
We were matching the direction case-insensitively in the regex, but then comparing it case-sensitively afterwards, so that lowercase S/W coordinates would not be parsed as S/W and would instead eventually end up being interpreted as unsigned (i. e. N/E). To fix this, it’s sufficient to check if the regex matched that part of the pattern at all or fell back to the empty alternative branch.

Credit for investigating the bug goes to [Wikidata user Nikki][1] [on Phabricator][2].

[1]: https://www.wikidata.org/wiki/User:Nikki
[2]: https://phabricator.wikimedia.org/T173026#4465312

Bug: [T173026](https://phabricator.wikimedia.org/T173026)

---

Note: Wikidata currently uses version 2.1.1 of this library, and I don’t think we could upgrade to 4.0.0 even if we wanted to (I don’t think we can use scalar type hints in production), so we’ll need to backport this as well. What’s the procedure for backports in this project?

### CODE REVIEW

File: src/Parsers/LatLongParserBase.php [hunk 1]
```diff
@@ -224,7 +224,7 @@ protected function resolveDirection( string $coordinateSegment ): string {
 				$matches
 			);
 
-			if ( $matches[1] === $direction || $matches[3] === $direction ) {
+			if ( $matches[1] !== '' || $matches[3] !== '' ) {
```
Comment (JeroenDeDauw) [inline src/Parsers/LatLongParserBase.php]: Looks like now it won't fail every time it should though. Better add some tests for that.

**Comment by JeroenDeDauw** [comment]
Weird 2.x is still used since there is a 3.x :)

To backport please make a PR into to the 3.0.x branch. Then make the 3.0.1 tag on that branch. If you want to go with 2.x, then make a 2.1.x branch based off the 2.1.1 tag. In that case you need to not forget to update the then still existing `DATAVALUES_GEO_VERSION` constant.

The README will also need to be updated. I don't really care about the readme on the branch, so update there is optional. The release should be in the release notes list on the master branch though. And there follow chronological order of release dates (so latest release, even if it is a backport, at the top).

File: src/Parsers/LatLongParserBase.php (hunk 1 above)
Comment (lucaswerkmeister) [reply src/Parsers/LatLongParserBase.php]: #139 only touches the tests as far as I can see, so do I understand correctly that the code works as it should after all?

Comment (JeroenDeDauw) [reply src/Parsers/LatLongParserBase.php]: Indeed.

**Comment by lucaswerkmeister** [comment]
Should I merge the pull requests myself or let someone else do that?

**Comment by JeroenDeDauw** [comment]
Are you able to merge them yourself? Normally someone else does it. No strong opinion on that but at the very least someone should double check either the PR or the state of things before the tag is created. Cause making a messed up tag can't easily be undone.

**Comment by lucaswerkmeister** [comment]
I’m probably able to merge them myself, but it sounds like I shouldn’t do it?

**Comment by JeroenDeDauw** [comment]
If you do that then please have someone double check everything is OK before tagging.

**Comment by JeroenDeDauw** [comment]
Anyway I merged them now

**Comment by lucaswerkmeister** [comment]
I’m not sure what the double checking means, beyond what we already have CI for… I checked manually that using the 2.1.x branch fixes the problem that was reported in Wikibase.

**Comment by JeroenDeDauw** [comment]
CI won't catch forgetting to update the version constant or cherry-picking the wrong commit onto the branch and then tagging it.

**Pull request merged**